explanations/best-practices/code-reviews.rst

```diff
@@ -0,0 +1,41 @@
+============
+Code reviews
+============
+
+Writing code is a craft. There is no absolute right or wrong way of expressing thoughts into code. It's mainly contextual:
+
+- What the current needs?
+- What are the current constraints?
+- What is our current understanding of the problem?
+
+Reviewing code is also contextual. Here below are some guidelines. They must not be applied in every circumstance. Instead, they define a baseline in terms of expectations on the team.
+
+Requesting a review
+===================
+
+- **Timing:** expect the reviewer to come back after a day or two, in order to take timezones into account.
+- **Load:** requests should be well balanced among team members so we don't overload a single individual.
+- **Reminders:** the Slack channel `#releng-notifications <https://app.slack.com/client/T027LFU12/CN77C0BUG>`__ will remind the reviewer when there is an outstanding request.
+- **Cognitive load:** it's easier for the human brain to focus on several things, one by one, rather than one big thing. Split your review into several atomic commits. That will speed up the review.
+- **Context:** provide information about why you are proposing this change.
+
+  - when the bug report is straightforward, in the commit message and the Pull Request title
+  - when the bug report requires lots of back-and-forth between people, on the bug report itself.
+
+- **All green:** ensure required checks are passing before submitting review request by creating
+
+  - the Pull Request as Draft
+  - or the Phabricator revision as "Changes planned".
+
+
+Performing a review
+===================
+
+- **Tone:** remain considerate. We all make mistakes and we can all learn from each other.
+- **Context:** feel free to ask more information from the author.
```

Comment:
My personal pet peeve is when a reviewer blocks waiting for more context but doesn't "Request changes". So I suggest adding something like:

> When requesting additional context, ensure you "Request changes". This puts the revision back in the author's queue and thereby helps avoid scenarios where they missed your question and the revision sits in limbo with both parties waiting for the other to respond.

Reply:
I've never encountered a case where someone missed my review because I didn't hit "request changes". I know some people prefer just commenting on the PR in order to keep the conversation going. I'm not sure we need to make "request changes" a guideline for now. I'm happy to discuss this, though 🙂

Reply:
It seems to happen on Phabricator more in my experience. And sure we can discuss more before including it.

Reply:
Big +1 to this. Keeping the official "state" of the PR or revision in sync with actual expectations helps keep things running smoothly. (We should also make sure that everyone knows that "request changes" or similar is not a criticism or failing - just a reflection of where things are.)

Reply:
Okay. I just added: 

> Use "Request Changes" to put the review back in the author's queue.